DBZ-7697 When new tables are added to MSSQL, its schema must be not captured.

Mon-MFEC opened the pull request:
The current Debezium captured the schema of newly added tables on MSSQL after initial deployment. This pull request will make that if a new tables are added on MSSQL, its schema must be not captured which be a follow the behavior as documented.

Mon-MFEC commented:
@jpechane  @mfvitale 
pls review. 😁

Mon-MFEC commented on debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerChangeTableSetIT.java:
@Naros 
Based on your suggestion in this [topic](https://debezium.zulipchat.com/#narrow/stream/348252-community-sqlserver/topic/Adding.20a.20table.20.20after.20deploy.20debezium/near/430239435),I think it's the test case: readHistoryAfterRestart.

Naros commented:
Please take a look at the test failures, I believe several are related to this change @Mon-MFEC .

Mon-MFEC commented:
I have tested as [this ](https://debezium.zulipchat.com/#narrow/stream/348252-community-sqlserver/topic/Adding.20a.20table.20.20after.20deploy.20debezium/near/430039341)

Naros commented:
Yes, but there are other failures in the test suite now that did not appear before.  Can you please confirm whether those failures are related to your code changes, thanks @Mon-MFEC .

Mon-MFEC commented:
I will try testing in the main branch, which has no code changes, and compare it with the DBZ-7697 branch, which is a branch that I modified the code myself (I will try testing only in the debezium-connector-sqlserver module), OK? @Naros

Mon-MFEC commented:
Hi @Naros @jpechane @mfvitale 
OK, just to make sure that my code changes don't affect other parts of Debezium, I'll try testing in the main branch first, when all test in the main branch pass then I'll testing in the my branch

1. I created a VM on GCP and set up the OS as RHEL 8.
2. I set up Docker and launched the MSSQL service (firewall allowed) with the following YAML file.
![docker-yml](https://github.com/debezium/debezium/assets/125636838/de91e45d-6ef9-4fbc-9bf0-0c110940f928)

3. In the Debezium project, I switched to the main branch and updated the code to the latest version.
4. After that, I made code adjustments in the package: **io.debezium.connector.sqlserver.util**, class **TestHelper**, method: **defaultJdbcConfig**, from the left configuration to the right configuration (the reason for the change is to align the configurations with the environment I prepared in steps 1 and 2).
![Code-Change](https://github.com/debezium/debezium/assets/125636838/d06312d8-4521-4d79-a8a8-f40b1963bf5d)
5. I ran all tests in the **debezium-connector-sqlserver** module, and the results showed some failing tests, as follows:

package : **io.debezium.pipeline**, Class : **AbstractBlockingSnapshotTest**, Test : **executeBlockingSnapshotWhileStreaming**
![error_1](https://github.com/debezium/debezium/assets/125636838/efa12690-447b-4e7f-9e7c-7d635eec614c)
[error_1.txt](https://github.com/debezium/debezium/files/14888810/error_1.txt)

 package **io.debezium.connector.sqlserver** , Class **SqlServerConnectorIT**, Test : **shouldNotStreamWhenUsingSnapshotModeInitialOnly** 
![error_2](https://github.com/debezium/debezium/assets/125636838/f2a7ddfe-3c3e-4f0a-b039-6a3b1f024109)
[error_2.txt](https://github.com/debezium/debezium/files/14888812/error_2.txt)

package **io.debezium.connector.sqlserver** , Class **SqlServerConnectorIT**, Test : **testEmptySchemaWarningAfterApplyingFilters**
![error_3](https://github.com/debezium/debezium/assets/125636838/15370b49-750e-49eb-891d-832ebde18ca9)
[error_3.txt](https://github.com/debezium/debezium/files/14888819/error_3.txt)


Do you think I need to adjust anything additionally to pass all the test in the main branch?
(Sorry for the late response. I've been busy lately, but I'll try to get back to you as soon as possible.)

Mon-MFEC commented:
Additionally, I encountered another failed test case as follows:
**package io.debezium.connector.sqlserver, Class SqlServerChangeTableSetIT, Test : addColumnToTableWithParallelWrites** 

```
public void addColumnToTableWithParallelWrites() throws Exception {
        final int RECORDS_PER_TABLE = 20;
        final int TABLES = 2;
        final int ID_START_1 = 10;
        final int ID_START_2 = 100;
        final int ID_START_3 = 1000;
        final Configuration config = TestHelper.defaultConfig()
                .with(SqlServerConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NO_DATA)
                .build();

        start(SqlServerConnector.class, config);
        assertConnectorIsRunning();
        TestHelper.waitForSnapshotToBeCompleted();

        for (int i = 0; i < RECORDS_PER_TABLE; i++) {
            final int id = ID_START_1 + i;
            connection.execute(
                    "INSERT INTO tablea VALUES(" + id + ", 'a')");
            connection.execute(
                    "INSERT INTO tableb VALUES(" + id + ", 'b')");
        }

        SourceRecords records = consumeRecordsByTopic(RECORDS_PER_TABLE * TABLES);
        assertThat(records.recordsForTopic("server1.testDB1.dbo.tablea")).hasSize(RECORDS_PER_TABLE);
        assertThat(records.recordsForTopic("server1.testDB1.dbo.tableb")).hasSize(RECORDS_PER_TABLE);
        records.recordsForTopic("server1.testDB1.dbo.tableb").forEach(record -> {
            assertSchemaMatchesStruct(
                    (Struct) ((Struct) record.value()).get("after"),
                    SchemaBuilder.struct()
                            .optional()
                            .name("server1.testDB1.dbo.tableb.Value")
                            .field("id", Schema.INT32_SCHEMA)
                            .field("colb", Schema.OPTIONAL_STRING_SCHEMA)
                            .build());
        });

        Executors.newSingleThreadExecutor().submit(() -> {
            try (JdbcConnection connection = TestHelper.testConnection()) {
                for (int i = 0; i < RECORDS_PER_TABLE; i++) {
                    final int id = ID_START_2 + i;
                    connection.execute(
                            "INSERT INTO tablea VALUES(" + id + ", 'a2')");
                    connection.execute(
                            "INSERT INTO tableb(id,colb) VALUES(" + id + ",'b')");
                    Thread.sleep(1000);
                }
            }
            catch (Exception e) {
                e.printStackTrace();
                throw new IllegalArgumentException(e);
            }
        });

        // Enable a second capture instance
        connection.execute("ALTER TABLE dbo.tableb ADD colb2 VARCHAR(32)");
        TestHelper.enableTableCdc(connection, "tableb", "after_change");

        records = consumeRecordsByTopic(RECORDS_PER_TABLE * 2);
        assertThat(records.recordsForTopic("server1.testDB1.dbo.tablea")).hasSize(RECORDS_PER_TABLE);
        assertThat(records.recordsForTopic("server1.testDB1.dbo.tableb")).hasSize(RECORDS_PER_TABLE);

        final AtomicInteger beforeChangeCount = new AtomicInteger();
        final AtomicInteger afterChangeCount = new AtomicInteger();
        records.recordsForTopic("server1.testDB1.dbo.tableb").forEach(record -> {
            if (((Struct) record.value()).getStruct("after").schema().field("colb2") != null) {
                afterChangeCount.incrementAndGet();
            }
            else {
                beforeChangeCount.incrementAndGet();
                assertThat(afterChangeCount.intValue()).isZero();
            }
        });
        assertThat(beforeChangeCount.intValue()).isPositive();
        assertThat(afterChangeCount.intValue()).isPositive();

        for (int i = 0; i < RECORDS_PER_TABLE; i++) {
            final int id = ID_START_3 + i;
            connection.execute(
                    "INSERT INTO tablea VALUES(" + id + ", 'a3')");
            connection.execute(
                    "INSERT INTO tableb VALUES(" + id + ", 'b1', 'b2')");
        }
        records = consumeRecordsByTopic(RECORDS_PER_TABLE * 2);
        assertThat(records.recordsForTopic("server1.testDB1.dbo.tablea")).hasSize(RECORDS_PER_TABLE);
        assertThat(records.recordsForTopic("server1.testDB1.dbo.tableb")).hasSize(RECORDS_PER_TABLE);
        records.recordsForTopic("server1.testDB1.dbo.tableb").forEach(record -> {
            assertSchemaMatchesStruct(
                    (Struct) ((Struct) record.value()).get("after"),
                    SchemaBuilder.struct()
                            .optional()
                            .name("server1.testDB1.dbo.tableb.Value")
                            .field("id", Schema.INT32_SCHEMA)
                            .field("colb", Schema.OPTIONAL_STRING_SCHEMA)
                            .field("colb2", Schema.OPTIONAL_STRING_SCHEMA)
                            .build());
        });
    }
```

I believe the cause of this is [this](https://debezium.zulipchat.com/#narrow/stream/348252-community-sqlserver/topic/Adding.20a.20table.20.20after.20deploy.20debezium/near/430042086)

When I changed the code in that test case from `assertThat(beforeChangeCount.intValue()).isPositive();` to `assertThat(beforeChangeCount.intValue()).isZero(); ` the result was that the test case passed successfully.

Mon-MFEC commented:
About **error_4**, I think we should include both of these conditions (`assertThat(beforeChangeCount.intValue()).isPositive()` and `assertThat(beforeChangeCount.intValue()).isZero()`) because I believe there is a chance that **beforeChangeCount** could have a value of both 0 and greater than 0. @Naros @jpechane @mfvitale
